fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -1612,6 +1750,95 @@ private static void parseInclusionParameter(Class<?> resourceType, FHIRSearchCon
         }
     }
 
+    /**
+     * Retrieves the search include restrictions.
+     *
+     * @param resourceType
+     *            the resource type
+     * @return list of allowed search _include values, or null if no restrictions
+     * @throws Exception
+     *             an exception
+     */
+    private static List<String> getSearchIncludeRestrictions(String resourceType) throws Exception {
+
+        // Retrieve the "resources" config property group.
+        PropertyGroup rsrcsGroup = FHIRConfigHelper.getPropertyGroup(FHIRConfiguration.PROPERTY_RESOURCES);
+        if (rsrcsGroup != null) {
+            List<PropertyEntry> rsrcsEntries = rsrcsGroup.getProperties();
+            if (rsrcsEntries != null && !rsrcsEntries.isEmpty()) {
+
+                // Try to find search includes property for matching resource type
+                for (PropertyEntry rsrcsEntry : rsrcsEntries) {
+                    if (resourceType.equals(rsrcsEntry.getName())) {
+                        PropertyGroup resourceTypeGroup = (PropertyGroup) rsrcsEntry.getValue();
+                        if (resourceTypeGroup != null) {
+                            return resourceTypeGroup.getStringListProperty(FHIRConfiguration.PROPERTY_FIELD_RESOURCES_SEARCH_INCLUDES);
+                        }
+                    }
+                }
+
+                // Otherwise, try to find search includes property for "Resource" resource type
+                for (PropertyEntry rsrcsEntry : rsrcsEntries) {
+
+                    // Check if matching resource type
+                    if (SearchConstants.RESOURCE_RESOURCE.equals(rsrcsEntry.getName())) {
+                        PropertyGroup resourceTypeGroup = (PropertyGroup) rsrcsEntry.getValue();
+                        if (resourceTypeGroup != null) {
+                            return resourceTypeGroup.getStringListProperty(FHIRConfiguration.PROPERTY_FIELD_RESOURCES_SEARCH_INCLUDES);
+                        }
+                    }
+                }
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Retrieves the search revinclude restrictions.
+     *
+     * @param resourceType
+     *            the resource type
+     * @return list of allowed search _revinclude values, or null if no restrictions
+     * @throws Exception
+     *             an exception
+     */
+    private static List<String> getSearchRevIncludeRestrictions(String resourceType) throws Exception {
```

Comment:
Please refactor this method and the prior into one processing logic method, and a new signature, resourceType, String propertyType, and leave the existing methods only pointing to the one logic method.

Reply:
Updated.